### ISSUE TITLE
Using S'UP for 1-on-1's

### ISSUE THREAD

**Issue by ancord-digital** [dblock/slack-sup#109]
It would be nice if we could use S'UP for 1-on-1 sessions as well. For example in our case where the CTO wants to do a check-in with all the developers every X months. 

The main requirements are: 

- there will be 1 main person who is always in the session, in this case our CTO, or a manager - the 'session lead'
- the developers should be able to opt-in - so you can create a team
- setting the schedule to meet everyone once per X months

**Comment by dblock** [comment]
This is tangentially related to being able to create a S'Up per channel #14. 

I suppose it's a good idea to say "sup helps a very large number of people opt into meeting 1:1 with a very important person without overwhelming their calendar". So how would this work?

1. CTO does `sup me` which establishes a S'Up 
2. CTO can do `sup me set size 1` to say that they want a sup with 1 person instead of 3 by default
3. CTO can do `sup me set weeks 1` to say that they want a new round of sups to be scheduled every week
4. CTO can do `sup me set recency 12` to say that they don't want to meet with someone more frequently than 12 weeks
4. Dev does `sup @cto opt in` to opt themselves into into CTO's S'Up
5. Every week S'Up DMs pairs of CTO + dev to schedule a 1:1

I think we'll need a new `set max` to limit the total number of S'Ups at any given week since these will always include the same person.

WDYT @ancord-digital ? Other ideas? What am I missing?

**Comment by dblock** [comment]
Thanks @gsomoza. Some level of control over opt in/out is needed, maybe `sup me set opt none` (users cannot opt in or out themselves) vs. `sup me set opt in,out` (users can both opt in and out) could  enable CTO's S'Ups opt in/out options by the users. That said, maybe users should always be able to opt out themselves from anything.

**Comment by gsomoza** [comment]
You're right: if admins can themselves add a user into an arbitrary S'Up, then I don't mind if the user can later opt themselves out of that – but in that case it would be nice to be notified so we can follow-up on why that happened.
